libs/core/execution_base/include/hpx/execution_base/traits/coroutine_traits.hpp

```diff
@@ -0,0 +1,231 @@
+//  Copyright (c) 2022 Shreyas Atre
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <coroutine>
+#include <type_traits>
+
+namespace hpx { namespace execution { namespace experimental {
+
+    template <typename Promise, typename Awaiter>
+    decltype(auto) await_suspend(Awaiter&& await)
+    {
+        if constexpr (!std::is_same_v<Promise, void>)
+        {
+            return await.await_suspend(std::coroutine_handle<Promise>{});
+        }
+        return;
+    }
+
+    namespace detail {
+
+        template <typename T, typename... Ts>
+        bool inline constexpr one_of = (std::is_same_v<T, Ts> || ...);
```

Comment:
We already have a similar facility here: https://github.com/STEllAR-GROUP/hpx/blob/84801de6a0003336287f2e4dd3da7138352c419f/libs/core/type_support/include/hpx/type_support/pack.hpp#L122-L140

Reply:
I think it's different
one_of is true if any one of the `Ts` matches with the given `T`. `any_of` only checks whether `is_false<Ts>` right?
Context HPX
```
util::any_of<is_one_way_executor<T>,
            is_never_blocking_one_way_executor<T>, is_bulk_one_way_executor<T>,
            is_two_way_executor<T>, is_bulk_two_way_executor<T>>
```
Where as I want to check if `is_same<int,bool,void>` (false) `is_same<bool, bool,void>` (true) `is_same<void, bool, void>` (true) ...

Reply:
FWIW, there also is `is_any_executor<T>` that resolves to true if any of the above conditions are true.

Reply:
I tried to find `is_any_executor` but could not.

Reply:
Sorry, it's `hpx::traits::is_executor_any<T>`.